lib/modules/datasource/docker/index.ts

```diff
@@ -70,15 +67,13 @@ export async function getAuthHeaders(
         // TODO: add cache test
         await http.getJson(apiCheckUrl, options);
 
-    if (apiCheckResponse.statusCode === 200) {
+    if (
+      apiCheckResponse.statusCode === 200 ||
+      apiCheckResponse.statusCode === 405
```

Comment:
should only be allowed when apicheckurl doesn't end with `/v2/`.

Reply:
@viceice in our case apicheckUrl is https://ecr-proxy.company.com/v2/node/tags/list?n=10000 and we are getting a 405 response from the un-authenticated proxy from the gttp.getJson Call.

Reply:
@viceice due to the fact that the proxy is handling the authentication for us to AWS ECR.

Reply:
Not changing given my investigation.